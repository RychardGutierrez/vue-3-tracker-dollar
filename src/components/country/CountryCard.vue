<template>
  <div v-if="country"
    class="flex py-4 px-4 bg-dollar-primary hover:bg-green-600 text-white rounded-md shadow-md cursor-pointer">
    <div class="flex flex-col flex-1">
      <h2 class="text-2xl">{{ country.official }}</h2>
      <h3>{{ country.savedDate }}</h3>
    </div>
    <div class="flex flex-col gap-2">
      <p class="text-3cl self-end">{{ country.rate }} {{ country.symbol }}</p>
      <p class="text-3cl self-end">
        {{ country.defaultRate }}
      </p>
    </div>
  </div>
</template>

<script setup>
import { defineProps } from 'vue';

// Props
defineProps({
  country: {
    type: Object,
    default: null,
  },
});
</script>
