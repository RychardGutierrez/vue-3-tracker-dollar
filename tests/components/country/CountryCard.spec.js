import { shallowMount } from '@vue/test-utils';
import CountryCard from '../../../src/components/country/CountryCard.vue';

describe('CountryCard component', () => {
  test('should to match with snapshot', () => {
    const defaultProps = {
      country: {
        official: 'Plurinational State of Bolivia',
        savedDate: 'Feb 5, 2023',
        rate: '6.95361',
        symbol: 'BO',
        defaultRate: '1 USD',
      },
    };
    const wraper = shallowMount(CountryCard, { props: defaultProps });

    expect(wraper.html()).toMatchSnapshot();
  });
});
