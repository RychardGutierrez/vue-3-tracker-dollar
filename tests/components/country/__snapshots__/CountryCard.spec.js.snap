// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard component should to match with snapshot 1`] = `
"<div class=\\"flex py-4 px-4 bg-dollar-primary hover:bg-green-600 text-white rounded-md shadow-md cursor-pointer\\">
  <div class=\\"flex flex-col flex-1\\">
    <h2 class=\\"text-2xl\\">Plurinational State of Bolivia</h2>
    <h3>Feb 5, 2023</h3>
  </div>
  <div class=\\"flex flex-col gap-2\\">
    <p class=\\"text-3cl self-end\\">6.95361 BO</p>
    <p class=\\"text-3cl self-end\\">1 USD</p>
  </div>
</div>"
`;
